x/ante/batch.go

```diff
@@ -0,0 +1,90 @@
+package ante
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	batchtypes "github.com/axelarnetwork/axelar-core/x/batch/types"
+)
+
+var _ sdk.FeeTx = (*BatchDecorator)(nil)
+
+// BatchDecorator implements the Tx interface and runs anteHandler on the inner messages of a batch request
+type BatchDecorator struct {
+	sdk.FeeTx
+
+	cdc      codec.Codec
+	messages []sdk.Msg
+}
+
+// NewBatchDecorator is the constructor for BatchDecorator
+func NewBatchDecorator(cdc codec.Codec) BatchDecorator {
+	return BatchDecorator{
+		nil,
+		cdc,
+		[]sdk.Msg{},
+	}
+}
+
+// AnteHandle record qualified refund for the multiSig and vote transactions
+func (b BatchDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (sdk.Context, error) {
+	msgs := tx.GetMsgs()
+
+	if !anyBatch(msgs) {
+		return next(ctx, tx, simulate)
+	}
+
+	feeTx, ok := tx.(sdk.FeeTx)
+	if !ok {
+		return ctx, sdkerrors.Wrap(sdkerrors.ErrTxDecode, "tx must be a FeeTx")
+	}
+
+	b.FeeTx = feeTx
+
+	for _, msg := range msgs {
+		switch req := msg.(type) {
+		case *batchtypes.BatchRequest:
+			innerMsgs := req.UnwrapMessages()
+			if anyBatch(innerMsgs) {
+				return ctx, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "nested batch requests are not allowed")
+			}
+
+			b.messages = append(b.messages, innerMsgs...)
+
+		default:
+			b.messages = append(b.messages, msg)
+		}
+
+	}
```

Comment:
I don't really get it, why is this split from the case when there is no batch? The `anyBatch` check seems kind of unnecessary